x-pack/legacy/plugins/integrations_manager/common/routes.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { PLUGIN_ID } from './constants';
+
+export const API_ROOT = `/api/${PLUGIN_ID}`;
+
+export enum RouteName {
+  API_LIST = 'API_LIST',
+  API_INFO = 'API_INFO',
+  API_ZIP = 'API_ZIP',
+  API_TGZ = 'API_TGZ',
+}
+
+type RouteNameKey = keyof typeof RouteName;
+type RouteMap = Record<RouteNameKey, RouteDef>;
+
+interface RouteDef {
+  path: string;
+  generatePath: (replacement?: string) => string;
```

Comment:
I should note this is “works for the 3 cases we have now”-level code. We’ll come back to this and change the shape as required.